aiidalab_widgets_base/wizard.py

```diff
@@ -118,10 +118,11 @@ def __init__(self, steps, **kwargs):
 
         def spinner_thread():
             while True:
-                sleep(0.1)
+                time.sleep(0.1)
                 self._update_titles()
 
-        Thread(target=spinner_thread).start()
+        if not testing:  # pytest is hanging if we start a thread
+            threading.Thread(target=spinner_thread).start()
```

Comment:
```suggestion
        self.spinner_thread = threading.Thread(target=spinner_thread).start()
```

The thread can be stop and joined in the test explicitly. It is also good to leave a handler attribute to get control of inner thread.
You

Reply:
Did you try `self.spinner_thread.join()` this will wait for the thread to stop and finish. We did it here https://github.com/aiidalab/aiidalab/blob/50498b08ea77937cb8af0e8d753446dd17b6d637/tests/test_appclass.py#L87-L104

Reply:
> Did you try `self.spinner_thread.join()` this will wait for the thread to stop and finish. We did it here https://github.com/aiidalab/aiidalab/blob/50498b08ea77937cb8af0e8d753446dd17b6d637/tests/test_appclass.py#L87-L104 we can try together.

The problem is that the current loop is infinite:

```python
while True:
    ....
```

So there is no reason for it to stop. And calling `_stop()` didn't stop it.